### PR TITLE
freefem: add missing dependencies

### PR DESCRIPTION
**Pull request by michaelkuhn**
Also remove the custom `autoreconf` method as it does not seem to be necessary and potentially hides useful output provided by the default.

### CODE REVIEW

**Review by becker33 — APPROVED** [review]

**Review by alecbcs — APPROVED** [review]
Looks good to me. @corentin-dev as maintainer do you have any notes before merge?

**Comment by SudhakarYogaraj** [comment]
I had the following error when installing Freefem@4.10 in Fedora 38, and this PR solved the problem.

==> Error: TypeError: 'NoneType' object is not callable
~/software/spack/var/spack/repos/builtin/packages/freefem/package.py:51, in autoreconf:
        49   def autoreconf(self, spec, prefix):
        50       autoreconf = which("autoreconf")
>>    51       autoreconf("-i")

**Review by corentin-dev — APPROVED** [review]

**Pull request merged**
